### PR TITLE
RK-3464 - use webpack to bundle code

### PR DESCRIPTION
**Pull request by amits1995**
introduced webpack to bundle dist files  

- [x] Need to address headless mode

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,26 +32,18 @@ jobs:
...
             # install dependencies
-            yarn
             # CircleCI sets CI=true which makes react return an error when there are warnings in the code
-            export CI=false
-            # build react app
-            yarn run build
-            # copy react bundle to dist directory
-            mv build/* ../../dist/
-            cd ../..
+            CI=false npm run --prefix src/webapp build
```
Comment (MickaelAlliel) [inline .circleci/config.yml]: yarn ?

File: webpack.index-worker.config.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+const path = require('path');
+
+const config = {
+  mode: 'development',
+  entry: './src/index-worker.ts',
+  target: "electron-renderer",
+  devtool: 'inline-source-map',
```
Comment (MickaelAlliel) [inline webpack.index-worker.config.js]: that's the reason we only got 10% out of it..

File: webpack.main.config.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+const path = require('path');
+
+const config = {
+  mode: 'production',
+  entry: './src/index.ts',
+  target: "electron-main",
+  devtool: "inline-source-map",
```
Comment (MickaelAlliel) [inline webpack.main.config.js]: that's the reason we only got 10% out of it..

File: src/server.ts [hunk 4]
```diff
@@ -59,7 +59,11 @@ export const start = (options: StartOptions): Promise<any> => {
 
   server.express.use(authorizationMiddleware(settings.accessToken));
   // tslint:disable-next-line:no-console
-  return server.start({ port: settings.port, formatError: (errors: any) => {
+  return server.start({
+      // fix webpack doesn't bundle subscriptions
+      subscriptions: false,
```
Comment (MickaelAlliel) [inline src/server.ts]: do we even have subscriptions?

File: package.json [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
+    "package-linux": "build --linux APPIMAGE",
+    "package-windows": "build --win",
+    "build-headless": "tsc",
     "dist": "build",
     "mac-publish": "export ELECTRON_BUILDER_ALLOW_UNRESOLVED_DEPENDENCIES=true && build --mac --win --linux -p always",
-    "watch": "tsc -w",
     "lint": "tslint -c tslint.json -p tsconfig.json",
-    "start": "yarn run build && cross-env development=1 electron ./dist/index.js",
+    "start": "cross-env development=1 yarn run build && cross-env development=1 electron ./dist/index.js",
     "debug": "yarn run build && cross-env development=1 electron --inspect-brk ./dist/index.js"
```
Comment (MickaelAlliel) [inline package.json]: missing cross-env development=1 before yarn run build here?

**Comment by MickaelAlliel** [comment]
& what with headless mode? what was needed?

**Comment by amits1995** [comment]
@MickaelAlliel  
THX for the CR  
About headless:  
I added the npm script `build-headless` which compiles the typescript files  
Once this goes in we will break FE tests but I dunno how else to do that, as I can't insert headles build to the build script...  
Any idea?

File: src/server.ts (hunk 4 above)
Comment (amits1995) [reply src/server.ts]: Nope... still a sad fix but that's all I have ATM lol

File: package.json (hunk 5 above)
Comment (amits1995) [reply package.json]: THX

File: webpack.main.config.js (hunk 3 above)
Comment (amits1995) [reply webpack.main.config.js]: Look at the end of the file

File: webpack.index-worker.config.js (hunk 2 above)
Comment (amits1995) [reply webpack.index-worker.config.js]: Look at the end of the file

File: .circleci/config.yml (hunk 1 above)
Comment (amits1995) [reply .circleci/config.yml]: Doesn't matter `yarn run` and `npm run` are equivalent

Comment (amits1995) [reply .circleci/config.yml]: Switched to `yarn` for consistency :)

**Pull request merged**